zerver/decorator.py

```diff
@@ -786,3 +790,40 @@ def _wrapped_view_func(request: HttpRequest, *args: Any, **kwargs: Any) -> HttpR
             return json_success()
         return view_func(request, *args, **kwargs)
     return _wrapped_view_func  # type: ignore # https://github.com/python/mypy/issues/1927
+
+def zulip_otp_required(view: Any=None,
+                       redirect_field_name: str='next',
+                       login_url: Optional[str]=None
+                       ) -> Callable[..., HttpResponse]:
+    """
+    The reason we need to create this function is that the stock
+    otp_required decorator doesn't play well with tests. We cannot
+    enable/disable if_configured parameter during tests since the decorator
+    retains its value due to closure.
+
+    Similar to :func:`~django.contrib.auth.decorators.login_required`, but
+    requires the user to be :term:`verified`. By default, this redirects users
+    to :setting:`OTP_LOGIN_URL`.
+    """
```

Comment:
Can you file an issue with `django_otp` about this?  It seems like they should be able to fix this with some refactoring so that they're checking the setting at runtime.

Reply:
Done. https://bitbucket.org/psagers/django-otp/issues/35/if_configured-in-otp_required-decorator.